crates/ruff/src/args.rs

```diff
@@ -155,10 +161,19 @@ pub struct CheckCommand {
     preview: bool,
     #[clap(long, overrides_with("preview"), hide = true)]
     no_preview: bool,
-    /// Path to the `pyproject.toml` or `ruff.toml` file to use for
-    /// configuration.
-    #[arg(long, conflicts_with = "isolated")]
-    pub config: Option<PathBuf>,
+    /// Either a path to a TOML configuration file (`pyproject.toml` or `ruff.toml`),
+    /// or a TOML `<KEY> = <VALUE>` pair (such as you might find in a `ruff.toml` config file)
+    /// overriding a specific config setting.
+    /// Overrides of individual settings using this option always take precedence
+    /// over all configuration files, including configuration files that were also
+    /// specified using `--config`.
```

Comment:
This is interesting, it's different than cargo's `--config` behavior. Why are we doing it this way instead of always applying in-order?

Reply:
Ah, I didn't realise that was cargo's behaviour!

I figured it would be easier for users to understand if overrides specified using `--config` worked as similarly as possible to the dedicated CLI flags such as `--line-length` etc — it's fewer concepts to hold in your head if they have ~basically the same rules. If cargo has different behaviour here I'm open to reconsidering this — though a lot of Python developers won't be familiar with cargo, so maybe they won't even notice the behaviour difference between our CLI and cargo!

The way I have it now is probably also slightly simpler in terms of the implementation, but that's obviously very much a secondary concern to the UX :)

Reply:
I guess this makes more sense with https://github.com/astral-sh/ruff/pull/9599/files#r1480577234 where we are limiting to a single config file. If we'll support more though I think "in-order" is the clearest way to talk about precedence.